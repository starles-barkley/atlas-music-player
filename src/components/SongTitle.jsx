export default function SongTitle() {
  return (
    <div>
      <p className="text-2xl">Tidal Drift</p>
      <p className="text-gray-500">Echoes of the Sea</p>
    </div>
  );
}