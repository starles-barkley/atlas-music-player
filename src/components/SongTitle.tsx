import Skeleton from 'react-loading-skeleton';
import 'react-loading-skeleton/dist/skeleton.css';
import { Song } from './MusicPlayer.tsx';

type songTitleProps = {
    loading: Boolean;
    playlist: Song[]
    currentSong: number
}

export function SongTitle({loading, playlist, currentSong}: songTitleProps) {
    return (
        <div className="flex flex-col my-2">
            {loading ? (
                <>
                    <Skeleton className="w-3/4 h-10 mb-2" />
                    <Skeleton className="w-1/2 h-6" />
                </>
            ) : (
                <>
                    <p className='font-bold text-4xl'>{playlist[currentSong].title}</p>
                    <p className='opacity-50 text-xl my-2'>{playlist[currentSong].artist}</p>
                </>
            )}
        </div>
    );
}