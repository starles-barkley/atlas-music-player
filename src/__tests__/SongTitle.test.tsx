import "@testing-library/jest-dom";
import { render, screen, cleanup } from "@testing-library/react";
import { expect, test, afterEach } from "vitest";
import SongTitle from "../components/SongTitle.tsx";

afterEach(() => {
    cleanup();
});


test("renders song title and artist correctly", () => {
    const { container } = render(<SongTitle title="Painted in Blue" artist="Soul Canvas"/>);

    expect(screen.getByText("Painted in Blue")).toBeInTheDocument();
    expect(screen.getByText("Soul Canvas")).toBeInTheDocument();

    expect(container).toMatchSnapshot();
});