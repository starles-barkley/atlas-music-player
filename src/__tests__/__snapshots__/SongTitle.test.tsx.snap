// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SongTitle > matches snapshot 1`] = `
<div>
  <div>
    <p
      class="font-comfortaa text-2xl text-text"
    >
      Test Song
    </p>
    <p
      class="font-montserrat text-text"
    >
      Test Artist
    </p>
  </div>
</div>
`;
